sdk/eventgrid/azure-eventgrid/azure/eventgrid/_publisher_client.py

```diff
@@ -0,0 +1,57 @@
+#
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+from typing import TYPE_CHECKING, Sequence
+import json
+
+from azure.core import PipelineClient
+from msrest import Deserializer, Serializer
+
+if TYPE_CHECKING:
+    # pylint: disable=unused-import,ungrouped-imports
+    from typing import Any
+
+from ._models import CloudEvent, EventGridEvent, CustomEvent
+from ._helpers import _get_topic_hostname_only_fqdn, _get_authentication_policy
+from ._generated._event_grid_publisher_client import EventGridPublisherClient as EventGridPublisherClientImpl
+from . import _constants as constants
+
+class EventGridPublisherClient(object):
+    """EventGrid Python Publisher Client.
+
+    :param str topic_hostname: The topic endpoint to send the events to.
+    :param credential: The credential object used for authentication which implements SAS key authentication or SAS token authentication.
+    :type credential: Union[~azure.core.credentials.AzureKeyCredential, azure.eventgrid.EventGridSharedAccessSignatureCredential]
+    """
+
+    def __init__(self, topic_hostname, credential, **kwargs):
+        # type: (str, Union[AzureKeyCredential, EventGridSharedAccessSignatureCredential], Any) -> None
+
+        topic_hostname = _get_topic_hostname_only_fqdn(topic_hostname)
+
+        self._topic_hostname = topic_hostname
+        auth_policy = _get_authentication_policy(credential)
+        self._client = EventGridPublisherClientImpl(authentication_policy=auth_policy, **kwargs)
+
+    def send(self, events, **kwargs):
+        # type: (Union[List[CloudEvent], List[EventGridEvent], List[CustomEvent]], Any) -> None
+        """Sends event data to topic hostname specified during client initialization.
+
+        :param events: A list of CloudEvent/EventGridEvent/CustomEvent to be sent.
+        :type events: Union[List[models.CloudEvent], List[models.EventGridEvent], List[models.CustomEvent]]
+        :rtype: None
```

Comment:
Some places we have :rtype: None, other types we emit rtype when we don't return.  I don't recall any specific guidance here, but it seems to beg that question.
cc @lmazuel

Reply:
there should be an rtype - will go through the docstrings and fix them in a later pr

Reply:
Echoing my normal mantra of "fine with that; let's just make sure we have an issue to track it."

Reply:
https://github.com/Azure/azure-sdk-for-python/issues/13223